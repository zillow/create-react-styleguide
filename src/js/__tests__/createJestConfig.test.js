const { createJestConfig } = require('../..');

describe('createJestConfig', () => {
    it('shallow merges config', () => {
        const config = createJestConfig({
            setupFilesAfterEnv: [],
        });
        expect(config.setupFilesAfterEnv).toStrictEqual([]);
        expect(config).toMatchSnapshot();
    });

    it('can be called without arguments', () => {
        const config = createJestConfig();
        expect(config).toMatchSnapshot();
    });

    describe('mocked process.env', () => {
        const OLD_ENV = process.env;

        beforeEach(() => {
            jest.resetModules();
            process.env = { ...OLD_ENV };
        });

        afterEach(() => {
            process.env = OLD_ENV;
        });

        it('logs config with DEBUG=true', () => {
            process.env.DEBUG = 'true';
            const mock = jest.spyOn(global.console, 'log').mockImplementation(() => {});
            createJestConfig();
            expect(mock).toHaveBeenCalledTimes(1);
            expect(mock.mock.calls[0][0]).toBe('createJestConfig object:');
        });
    });
});
