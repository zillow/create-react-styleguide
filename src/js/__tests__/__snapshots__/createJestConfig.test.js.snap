// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createJestConfig can be called without arguments 1`] = `
Object {
  "collectCoverage": false,
  "collectCoverageFrom": Array [
    "src/**/*.js",
    "src/**/*.jsx",
  ],
  "coveragePathIgnorePatterns": Array [],
  "coverageReporters": Array [
    "cobertura",
    "html",
    "text",
  ],
  "coverageThreshold": Object {
    "global": Object {
      "branches": 75,
      "functions": 80,
      "lines": 70,
      "statements": 70,
    },
  },
  "setupFilesAfterEnv": Array [
    "<rootDir>/jest.setup.js",
  ],
  "testEnvironment": "node",
  "testMatch": Array [
    "<rootDir>/**/__tests__/**/*.test.js",
    "<rootDir>/**/*.test.js",
  ],
  "testPathIgnorePatterns": Array [
    "/node_modules/",
    "<rootDir>/lib/",
    "<rootDir>/es/",
  ],
}
`;

exports[`createJestConfig shallow merges config 1`] = `
Object {
  "collectCoverage": false,
  "collectCoverageFrom": Array [
    "src/**/*.js",
    "src/**/*.jsx",
  ],
  "coveragePathIgnorePatterns": Array [],
  "coverageReporters": Array [
    "cobertura",
    "html",
    "text",
  ],
  "coverageThreshold": Object {
    "global": Object {
      "branches": 75,
      "functions": 80,
      "lines": 70,
      "statements": 70,
    },
  },
  "setupFilesAfterEnv": Array [],
  "testEnvironment": "node",
  "testMatch": Array [
    "<rootDir>/**/__tests__/**/*.test.js",
    "<rootDir>/**/*.test.js",
  ],
  "testPathIgnorePatterns": Array [
    "/node_modules/",
    "<rootDir>/lib/",
    "<rootDir>/es/",
  ],
}
`;
